gui/src/types.ts

```diff
@@ -23,15 +23,26 @@ export const networkSchema = z.object({
 
 export type Network = z.infer<typeof networkSchema.shape.networks>[number];
 
+export type Address = `0x${string}`;
+
 export const walletSchema = z.object({
+  name: z.string().min(1),
+  dev: z.boolean().default(false),
   mnemonic: z.string().regex(/^(\w+\s){11}\w+$/, {
     message: "Must be a 12-word phrase",
   }),
   derivationPath: z.string().regex(/^m\/(\d+'?\/)+\d+$/, {
     message: "invalid path format",
   }),
   idx: z.number().int().min(0).max(4),
+  count: z.number().int().min(1),
+  currentKey: z.string().optional(),
```

Comment:
`currentKey` is an index? address? What is it?

Reply:
it's the index of which address is currently selected in this wallet

it's not part of the form, though. just needs to exist on the infered Wallet type

Reply:
to be honest don't like either of those, although the current one is also not great

my thinking is that `currentIndex` makes it sound like an array with numeric indexes, and `currentAccount` introduces yet another ambiguous word (wallet, address, account, it becomes very confusing)

the thinking behind key is that it's suposed to be agnostic about what that means inside the wallet, as if the wallet was a HashMap<key, Address>

Reply:
`currentPath`? Path is a normal word in derivation paths.

Reply:
renamed to `currentPath`